README.md

```diff
@@ -1,92 +1,93 @@
+## Testsuite
 
-Note: this is similar to [rvagg/abstract-leveldown](https://github.com/rvagg/abstract-leveldown/). Though I wrote [my original datastore](https://github.com/jbenet/datastore) many years ago. :)
+Available under `test/interface`
 
-## Example
+```js
+describe('mystore', () => {
+  require('interface-datastore/test/interface')({
+    setup (callback) {
+      callback(null, instanceOfMyStore)
+    },
+    teardown (callback) {
+      // cleanup resources
+      callback()
+    }
+  })
+})
+```
 
-### Usage
+## API
 
-See [datastore.memory/try.js](https://github.com/jbenet/node-datastore.memory/blob/master/try.js):
+The exact types can be found in [`src/index.js`](src/index.js).
 
-```js
-var memDS = require('datastore.memory')
-ds.put('foo', 'bar', function(err, val, key) {
-  if (err) throw err
-  console.log('put ' + key + ': ' + val)
-  assert(val === 'bar')
-})
+### `put(Key, Value, (err: ?Error) => void): void`
 
-ds.has('foo', function(err, has, key) {
-  if (err) throw err
-  console.log(key + ' exists? ' + has)
-  assert(has === true)
-})
+### `get(Key, (err: ?Error, val: ?Value) => void): void`
 
-ds.get('foo', function(err, val, key) {
-  if (err) throw err
-  console.log('get ' + key + ': ' + val)
-  assert(val === 'bar')
-})
+### `delete(Key, (err: ?Error) => void): void`
 
-ds.delete('foo', function(err, key) {
-  if (err) throw err
-  console.log(key + ' deleted')
-})
+### `query(Query<Value>): QueryResult<Value>)`
 
-ds.has('foo', function(err, has, key) {
-  if (err) throw err
-  console.log(key + ' exists? ' + has)
-  assert(has === false)
-})
-```
+#### `Query`
 
-### Implementation
+Object in the form with the following optional properties
 
-See [datastore.memory/index.js](https://github.com/jbenet/node-datastore.memory/blob/master/index.js):
+- `prefix?: string`
+- `filters?: Array<Filter<Value>>`
+- `orders?: Array<Order<Value>>`
+- `limit?: number`
+- `offset?: number`
+- `keysOnly?: bool`
 
-```js
-var DS = require('datastore.abstract')
-
-module.exports = MemDS
-
-function MemDS() {
-  if (!(this instanceof MemDS))
-    return new MemDS
-  DS.call(this)
-  this.values = {}
-}
-
-DS.inherits(MemDS)
-
-MemDS.prototype._get = function(key, cb) {
-  var val = this.values[key.toString()]
-  if (val !== undefined) cb(null, val, key)
-  else cb(MemDS.errors.NotFound, null, key)
-}
-
-MemDS.prototype._put = function(key, val, cb) {
-  this.values[key.toString()] = val
-  cb(null, val, key)
-}
-
-MemDS.prototype._delete = function(key, cb) {
-  delete this.values[key.toString()]
-  cb(null, key)
-}
-
-MemDS.prototype._has = function(key, cb) {
-  var has = (this.values[key.toString()] !== undefined)
-  cb(null, has, key)
-}
-```
+### batch(): Batch<Value>
+
+#### `put(Key, Value): void`
+#### `delete(Key): void`
+#### `commit((err: ?Error) => void): void`
+
+### `close((err: ?Error) => void): void`
 
+Close the datastore, this should always be called to ensure resources are cleaned up.
 
 ## License
 
-MIT
```

Comment:
The README needs a reference to https://github.com/ipfs/go-datastore and also, list out all the methods that are not implemented, if any.